### PR TITLE
Added documentation for Custom Events / `this.fire()`

### PR DESCRIPTION
**Pull request by gaurav21r**
There seems to be no documentation for custom events which was there in 0.5. Initial text, with style conventions of Polymer. Pl update if required.

Seems more people are confused, like me -> http://stackoverflow.com/questions/30544290/polymer-1-0-custom-events

### CODE REVIEW

**Comment by gaurav21r** [comment]
I signed it! Corporate CLA under company name `Ramanan Corporation`.

File: 1.0/docs/devguide/events.md [hunk 1]
```diff
@@ -102,6 +102,41 @@ So the attribute `on-core-signal-newData` sets up a listener for `core-signal-ne
...
+      </template>
+
+      <script>
+
+        Polymer({
+
+          is: 'x-custom',
+
+          handleClick: function() {
+            this.fire('kick', { kicked: true })
```
Comment (ebidel) [inline 1.0/docs/devguide/events.md]: spacing an `;`: `this.fire('kick', {kicked: true});`

File: 1.0/docs/devguide/events.md [hunk 2]
```diff
@@ -102,6 +102,41 @@ So the attribute `on-core-signal-newData` sets up a listener for `core-signal-ne
...
+        <button on-click="handleClick">Kick Me</button>
+      </template>
+
+      <script>
+
+        Polymer({
+
+          is: 'x-custom',
+
+          handleClick: function() {
```
Comment (ebidel) [inline 1.0/docs/devguide/events.md]: `function(e, detail) {`

File: 1.0/docs/devguide/events.md [hunk 3]
```diff
@@ -102,6 +102,41 @@ So the attribute `on-core-signal-newData` sets up a listener for `core-signal-ne
...
+            this.fire('kick', { kicked: true })
+          }
+
+        });
+
+      </script>
+
+    </dom-module>
+    
+    <x-custom> </x-custom>
```
Comment (ebidel) [inline 1.0/docs/devguide/events.md]: no space: `<x-custom></x-custom>`

File: 1.0/docs/devguide/events.md [hunk 4]
```diff
@@ -102,6 +102,41 @@ So the attribute `on-core-signal-newData` sets up a listener for `core-signal-ne
...
+    
+    <script>
+        document.querySelector('x-custom').addEventListener('kick', function (e){
```
Comment (ebidel) [inline 1.0/docs/devguide/events.md]: spacing: `function (e) {`

File: 1.0/docs/devguide/events.md [hunk 5]
```diff
@@ -102,6 +102,41 @@ So the attribute `on-core-signal-newData` sets up a listener for `core-signal-ne
...
+            console.log (e.kicked); //Prints true
```
Comment (ebidel) [inline 1.0/docs/devguide/events.md]: `console.log(e.kicked); // true`

**Comment by ebidel** [comment]
Reviewed

File: 1.0/docs/devguide/events.md [hunk 6]
```diff
@@ -102,6 +102,41 @@ So the attribute `on-core-signal-newData` sets up a listener for `core-signal-ne
 _not_ `core-signal-newData`. To avoid confusion, always use lowercase event names.
 {: .alert .alert-info } 
 
+## Custom Events {#custom-events}
+
+To fire a custom event from the host element use `this.fire()`. You can also pass in data to event handlers as an argument to `this.fire()`.
```
Comment (arthurevans) [inline 1.0/docs/devguide/events.md]: For consistency with other docs, this should be "the `fire` method." -- no `this.`, no parens unless you're listing arguments. (The second reference can just be `fire`.)

**Comment by gaurav21r** [comment]
@ebidel @arthurevans Thanks for the review :smiley: Will get right on it!

**Comment by gaurav21r** [comment]
@ebidel @arthurevans I've made the changes according to your review. Pl check.
Looking forward to getting a commit into **Polymer** :grinning: !!!

**Comment by ebidel** [comment]
LGTM

**Pull request merged**